pkg/package-server/provider/registry.go

```diff
@@ -189,7 +189,18 @@ func (p *RegistryProvider) syncCatalogSource(obj interface{}) (syncError error)
 		Namespace: source.GetNamespace(),
 		Name:      source.GetName(),
 	}
+
 	if sourceMeta := p.sources.GetMeta(key); sourceMeta != nil && sourceMeta.Address == address {
+		logger.Infof("updating PackageManifest based on CatalogSource changes: %v", key)
+		timeout, cancel := context.WithTimeout(context.Background(), cacheTimeout)
+		defer cancel()
+		client, err := p.registryClient(key)
+		if err != nil {
+			syncError = err
+			return
+		}
+		syncError = p.refreshCache(timeout, client)
+
 		// If the address hasn't changed, don't bother creating a new source
 		logger.Debug("catalog address unchanged, skipping source creation")
```

Comment:
I feel this debug line should be moved above the sync process. We should sync after we determine that catalog is unchanged right?

Reply:
you are right to pick on this, but I also thing this bugbug is nolonger needed since we are syncing and not skipping the updatae entirely.